icepyx/core/query.py

```diff
@@ -932,233 +297,218 @@ def avail_granules(self, ids=False, cycles=False, tracks=False, cloud=False):
+        files = earthaccess.download(links, local_path=path)
+        return files
 
-        if "email" in self._reqparams.fmted_keys or email is False:
-            self._reqparams.build_params(**self._reqparams.fmted_keys)
-        elif email is True:
-            user_profile = self.auth.get_user_profile()  # pyright: ignore[reportAttributeAccessIssue]
-            self._reqparams.build_params(
-                **self._reqparams.fmted_keys, email=user_profile["email_address"]
-            )
-
-        if subset is False:
-            self._subsetparams = None
-        elif (
-            subset is True
-            and hasattr(self, "_subsetparams")
-            and self._subsetparams is None
-        ):
-            del self._subsetparams
-
-        # REFACTOR: add checks here to see if the granules object has been created,
-        # and also if it already has a list of avail granules (if not, need to create one and add session)
-        if not hasattr(self, "_granules"):
-            self.granules
-
-        # Place multiple orders, one per granule, if readable_granule_name is used.
-        if "readable_granule_name[]" in self.CMRparams:
-            gran_name_list = self.CMRparams["readable_granule_name[]"]
-            tempCMRparams = self.CMRparams
-            if len(gran_name_list) > 1:
-                print(
-                    "NSIDC only allows ordering of one granule by name at a time; your orders will be placed accordingly."
-                )
-            for gran in gran_name_list:
-                tempCMRparams["readable_granule_name[]"] = gran
-                self._granules.place_order(
-                    tempCMRparams,
-                    cast(EGIRequiredParamsDownload, self.reqparams),
-                    self.subsetparams(**kwargs),
-                    verbose,
-                    subset,
-                    geom_filepath=self._spatial._geom_file,
-                )
-
-        else:
-            self._granules.place_order(
-                self.CMRparams,
-                cast(EGIRequiredParamsDownload, self.reqparams),
-                self.subsetparams(**kwargs),
-                verbose,
-                subset,
-                geom_filepath=self._spatial._geom_file,
-            )
+    def skip_preview(self):
+        """
+        skips preview for current order
+        """
+        if self.last_order and self.last_order.type == "subset":
+            status = self.last_order.status()
+            if status["status"] == "PREVIEW":
+                return self.last_order.resume()
 
-    # DevGoal: put back in the kwargs here so that people can just call download granules with subset=False!
-    def download_granules(
-        self, path, verbose=False, subset=True, restart=False, **kwargs
-    ):  # , extract=False):
+    def order_granules(
+        self, subset: bool = True, skip_preview: bool = False, path: str = "./"
+    ) -> DataOrder:
```

Comment:
to me it is confusing that this function "orders" granules if they're being subset, but "downloads" granules if they're not, and then there's another `download_granules` function that only "downloads" ordered+subset granules (but not the whole granules). Is there a functional reason not to just move the else steps from here to the `download_granules` function and do nothing if a user calls `order_granules` with `subset=False`?